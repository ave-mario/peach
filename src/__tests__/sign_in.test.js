import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SignIn from '../components/sign_in';
import faker from 'faker';

describe('SignIn form:', () => {
  Enzyme.configure({ adapter: new Adapter() });
  const fakePhoneNumber = faker.phone.phoneNumber;
  const props = {
    show: false,
    close: jest.fn(),
    attemtLogin: jest.fn(),
  };

  it('component renders correctly', () => {
    const SignInComponent = renderer.create(<SignIn {...props} />).toJSON();
    expect(SignInComponent).toMatchSnapshot();
  });

  it('component renders correctly', () => {
    let newProps = {
      ...props,
      show: true,
    };
    const SignInComponent = renderer.create(<SignIn {...newProps} />).toJSON();
    expect(SignInComponent).toMatchSnapshot();
  });

  it('component doesnt render by default', () => {
    const SignInComponent = mount(<SignIn {...props} />);
    expect(SignInComponent.prop('show')).toEqual(false);
  });

  it('inputs send value to state', () => {
    const SignInComponent = shallow(<SignIn {...props} />);
    SignInComponent.find('.phoneNumber').simulate('change', {
      target: { name: 'phoneNumber', value: fakePhoneNumber },
    });
    expect(SignInComponent.state('phoneNumber')).toEqual(fakePhoneNumber);
  });

  it('when form submited - prevent default', () => {
    const SignInComponent = shallow(<SignIn {...props} />);
    let prevented = false;
    SignInComponent.find('.formBody').simulate('submit', {
      preventDefault: () => {
        prevented = true;
      },
    });
    expect(prevented).toBe(true);
  });
});
