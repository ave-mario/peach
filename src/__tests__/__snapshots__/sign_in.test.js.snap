// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn form: component renders correctly 1`] = `
<div
  className="sc-htpNat lgeJtQ"
  style={
    Object {
      "opacity": "0",
      "transform": "translateY(-100px)",
    }
  }
>
  <div
    className="sc-bxivhb gPVRgU"
  >
    <h3
      className="sc-ifAKCX bXZmPy"
    >
      Login
    </h3>
    <span
      className="sc-bwzfXH goCgBH"
      onClick={[MockFunction]}
    >
      ×
    </span>
  </div>
  <form
    className="formBody sc-EHOje fFPbRn"
    onSubmit={[Function]}
  >
    <p>
      Please, enter your phone number:
    </p>
    <input
      className="phoneNumber sc-bZQynM bGXiwX"
      name="phoneNumber"
      onChange={[Function]}
      placeholder="email or phone number"
    />
  </form>
  <div
    className="sc-gzVnrw dphOmL"
  >
    <button
      className="sc-bdVaJa bTMOaQ"
      onClick={[MockFunction]}
    >
      CLOSE
    </button>
    <button
      className="sc-bdVaJa cFGEHa"
      onClick={[Function]}
      type="submit"
    >
      CONTINUE
    </button>
  </div>
</div>
`;

exports[`SignIn form: component renders correctly 2`] = `
<div
  className="sc-htpNat lgeJtQ"
  style={
    Object {
      "opacity": "1",
      "transform": "translateY(0vh)",
    }
  }
>
  <div
    className="sc-bxivhb gPVRgU"
  >
    <h3
      className="sc-ifAKCX bXZmPy"
    >
      Login
    </h3>
    <span
      className="sc-bwzfXH goCgBH"
      onClick={[MockFunction]}
    >
      ×
    </span>
  </div>
  <form
    className="formBody sc-EHOje fFPbRn"
    onSubmit={[Function]}
  >
    <p>
      Please, enter your phone number:
    </p>
    <input
      className="phoneNumber sc-bZQynM bGXiwX"
      name="phoneNumber"
      onChange={[Function]}
      placeholder="email or phone number"
    />
  </form>
  <div
    className="sc-gzVnrw dphOmL"
  >
    <button
      className="sc-bdVaJa bTMOaQ"
      onClick={[MockFunction]}
    >
      CLOSE
    </button>
    <button
      className="sc-bdVaJa cFGEHa"
      onClick={[Function]}
      type="submit"
    >
      CONTINUE
    </button>
  </div>
</div>
`;
